.github/workflows/spec-gen-sdk-status.yml

```diff
@@ -7,23 +7,47 @@ on:
 permissions:
   contents: read
   statuses: write
+  id-token: write
 
 jobs:
   sdk-validation-status:
     # Only run this job when the check run is from Azure Pipelines(SDK Validation) in the azure-sdk/public(id: 29ec6040-b234-4e31-b139-33dc4287b756) project
     if: |
       github.event.check_run.check_suite.app.name == 'Azure Pipelines' &&
-      contains(github.event.check_run.name, 'SDK Validation') &&
-      endsWith(github.event.check_run.external_id, '29ec6040-b234-4e31-b139-33dc4287b756')
-    name: "SDK Validation - Status"
+      contains(github.event.check_run.name, 'SDK Validation')
+    name: "SDK Validation Status"
     runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v4
         with:
           sparse-checkout: |
             .github
-      
-      - name: "SDK Validation - Status"
+
+      - name: Check Azure Credentials
+        id: check-creds
+        run: |
+          if [ -n "${{ secrets.AZURE_CLIENT_ID }}" ]; then
+            echo "has_creds=true" >> $GITHUB_OUTPUT
+          else
+            echo "has_creds=false" >> $GITHUB_OUTPUT
+          fi
+
+      - if: steps.check-creds.outputs.has_creds == 'true'
+        name: Azure Login with Workload Identity Federation
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
```

Comment:
Let's not have these be secrets as they are just ids that don't need to be protected.